includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+	 * @param array $rules array of the parent block settings.
+	 *
+	 * @return \stdClass Prepared post data.
+	 */
+	public function get_post_details( $post_ID, $rules ) {
+		$post_data = new \stdClass();
+
+		$post_data->postLink = get_the_permalink( $post_ID );
+		if ( $rules['display_featured_image'] ) {
+			$post_data->featuredImage = get_the_post_thumbnail( $post_ID );
+		}
+		if ( $rules['display_author'] ) {
+			$author_id             = get_post_field( 'post_author', $post_ID );
+			$author_display_name   = get_the_author_meta( 'display_name', $author_id );
+			$byline                = sprintf( __( 'by %s', 'sophi-wp' ), $author_display_name );
+			$post_data->postAuthor = $byline;
+		}
+		if ( $rules['display_post_date'] ) {
+			$post_data->postDate  = get_the_date( '', $post_ID );
+			$post_data->postDateC = get_the_date( 'c', $post_ID );
+		}
+		if ( $rules['display_post_excerpt'] ) {
+			$post_data->post_excerpt = wp_kses_post( get_the_excerpt( $post_ID ) );
+		}
+
+		return $post_data;
+	}
+
+	/**
+	 * Get the query params to get the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	public function site_automation_params() {
+		$params['pageName'] = array(
+			'description'       => __( 'Name of the page.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['widgetName'] = array(
+			'description'       => __( 'Name of the widget.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the get sophi posts query params.
+		 *
+		 * @param array $params Query params.
+		 *
+		 * @since 1.3.0
+		 */
+		return apply_filters( 'site_automation_params', $params );
+	}
+
+	/**
+	 * Update the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param  WP_REST_Request $request The request.
+	 * @return mixed
+	 */
+	public function site_automation_override( $request ) {
+		$current_user = wp_get_current_user();
+
+		if ( $current_user->exists() ) {
+			$user_email = $current_user->user_email;
+		} else {
+			return new \WP_Error( 401, __( 'Unauthorised user, please log in.', 'sophi-wp' ) );
+		}
+
+		// Get the auth token.
+		$api_token = $this->auth->get_access_token();
+
+		if ( is_wp_error( $api_token ) ) {
+			return new \WP_Error( 401, __( 'Invalid API token, please try adding correct credentials on the settings page.', 'sophi-wp' ) );
+		}
+
+		// Request parameters.
+		$timeout    = 3;
+		$attributes = $request->get_query_params();
+
+		$api_url = get_sophi_settings( 'sophi_override_url' );
+		$host    = get_sophi_settings( 'host' );
+		$api_url = $api_url . 'hosts/' . $host . '/overrides';
+
+		$rule_type        = isset( $attributes['ruleType'] ) ? sanitize_title( $attributes['ruleType'] ) : '';
```

Comment:
We have this pattern in a few places and this is totally optional to change but we could simplify this code a bit by using the null coalescing operator, so something like
```php
$attributes['ruleType'] ?? ''
```

We do lose the escaping there, but that could be moved to when we use these values in the `$body` below, which is also a little nicer because it's more clear that those values are escaped

Reply:
https://github.com/globeandmail/sophi-for-wordpress/pull/333/commits/86d6081ea65378dfa82a86b0c753ac542a491995